### ISSUE TITLE
Should package name map resolution allow overriding URLs?

### ISSUE THREAD

**Issue by domenic** [domenic/package-name-maps#49]
In particular, given a package name map like the following:

```json
{
  "packages": {
    "https://example.com/foo": { "main": "node_modules/src/foo.js" },
  }
}
```

and a JS file like

```js
import "https://example.com/foo";
```

should the result import the file from `https://example.com/foo`, or from `node_modules/src/foo.js`?

Stated in spec terms, should the package name map resolution come before or after step 1 of [resolve a module specifier](https://html.spec.whatwg.org/multipage/webappapis.html#resolve-a-module-specifier)?

I really like the extra power of allowing overriding of URLs. It does depart from the literal sense of "package name map" though, and might imply a renaming of this project to something more like "module resolution map" or similar.

**Comment by ljharb** [comment]
I’d expect a different top level key besides “packages” - maybe “urls” - but i also like the extra power.

**Comment by domenic** [comment]
Hmm yeah. I guess it kind of ties into previous discussions around how much we want to restrict package names.

**Comment by mkay581** [comment]
Hmm maybe we should wait until there is a strong use case for it? I can't seem to think of any.

**Comment by domenic** [comment]
The strong use case is written up at https://github.com/drufball/layered-apis/issues/34 :). I'll write a new issue with a proposed solution.
